mlx/fast.h

```diff
@@ -61,4 +62,52 @@ array affine_dequantize(
     int bits = 4,
     StreamOrDevice s = {});
 
+typedef std::variant<int, bool, Dtype> TemplateArg;
+
+class MetalKernel {
```

Comment:
Indeed it might be better to expose this as a function in the C++ API as it will simplify binding code. O/w we have to bind this class in MLX C to make it work (or special case this file).

Reply:
I ended up moving everything to a single function. I don't think the overhead of constructing the signature is going to matter much in practice and as you suggested, it unifies the Python/C++ APIs and makes the binding code a bit simpler.